### ISSUE TITLE
Safari 18.1.1 on MacOS 15.1.1 becomes unstable after some time (after waking up)  

### ISSUE THREAD

**Issue by StreborStrebor** [AdguardTeam/AdGuardMiniForMac#1040]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I made sure all content blockers are enabled in Safari settings
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

1.11.20

### Environment

```markdown
- OS: 15.1.1
- Safari version: 18.1.1 (20619.2.8.11.12)
- M3Max Macbook pro
```

### What filters do you have enabled?

AdGuard Social Media filter, AdGuard Other Annoyances filter

### Issue Details

Steps to reproduce:
1. Eneable Adguard Custom, Adguard for Safari, Adguard General, Adguard Other, Privacy, Security and Social extensions. I have not customised them.
2. Launch safari, use it with many tabs
3. After some time (and most certainly after waking up my M3 Max Macbook pro at a new location/network) safari starts behaving badly. Entering a URL in the address bar and hitting ENTER will do nothing (you have to click on the page and it will start loading). Opening URLS from mail, to open in Safari then crashes safari (beachball forever)
4. Only option now is to force quit

### Expected Behavior

Safari to always stay responsive (like it always was for me)

### Actual Behavior

- Safari eventually goes unresponsive.
- After some time safari starts behaving badly. Entering a URL in the address bar and hitting ENTER will do nothing (you have to click on the page and it will start loading). Opening URLS from mail, to open in Safari then crashes safari (beachball forever)
- Safari or no other processes are using CPU or hogging memory. It just becomes unresponsive.

### Screenshots

<details><summary>Screenshot 1:</summary>

</details>

### Additional Information

Disabling the Adguard extensions fixes the issue and Safari does not stop responding
This seemed to happen when MacOS 15.1.1 came out (I was running 15.1.0 and updated as soon as 15.1.1 came out, which included security updates, which could have broken something). 
This update might align with an update to Adblock (which could have also broken something).

Sorry I can't give more detail, but maybe others are noticing this too and can hook into this topic with more useful information.

**Comment by windwak3r** [comment]
@StreborStrebor, hello! How many profiles do you have in Safari?

**Comment by StreborStrebor** [comment]
Awfully sorry for this erroneous bug report.
It seems that this now also happens when I disable all Adguard extensions (So my next step will be to enable them again).
I have no other extensions enabled.

@windwak3r I have 20 profiles in Safari.
I did check that the Adguard extensions were disabled for each profile.

**Comment by StreborStrebor** [comment]
As mentionned above, this is an erroneous report, so closing it.
Keep up the good work!

**Comment by windwak3r** [comment]
@StreborStrebor the problem might be related to https://github.com/AdguardTeam/AdGuardForSafari/issues/1024 and somehow connected to having many profiles in Safari.
Does the problem reproduce if you uninstall AdGuard for Safari or reduce the number of profiles?

**Comment by StreborStrebor** [comment]
I don't know. I will deinstall Adguard and see if that fixes it.
I've had this amount of Safari profiles, since safari allowed profiles. I'd rather not reduce them (as I use them all).

**Comment by StreborStrebor** [comment]
> @StreborStrebor the problem might be related to #1024 and somehow connected to having many profiles in Safari. Does the problem reproduce if you uninstall AdGuard for Safari or reduce the number of profiles?

Yes, removing Adguard (eventhough the extensions were all disabled in Safari) fixed the problem.
Since removing Adguard, Safari has not crashed once.
Before removing Adguard, Safari crashed many times a day.

I've had ~ 18 profiles. But recently I added on or two more, now totalling 20.
